common/lib/xmodule/xmodule/library_tools.py

```diff
@@ -37,13 +38,40 @@ def get_library_version(self, lib_key):
         Get the version (an ObjectID) of the given library.
         Returns None if the library does not exist.
         """
-        library = self._get_library(lib_key)
+        library = self.get_library(lib_key)
         if library:
             # We need to know the library's version so ensure it's set in library.location.library_key.version_guid
             assert library.location.library_key.version_guid is not None
             return library.location.library_key.version_guid
         return None
 
+    def _filter_child(self, capa_type, child_descriptor):
+        """
+        Filters children by CAPA problem type, if configured
+        """
+        if capa_type == ANY_CAPA_TYPE_VALUE:
+            return True
+
+        if not isinstance(child_descriptor, CapaDescriptor):
+            return False
+
+        return capa_type in child_descriptor.problem_types
+
+    def get_filtered_children(self, from_block, capa_type=ANY_CAPA_TYPE_VALUE):
+        """
+        Filters children of `from_block` that satisfy filter criteria
+        Returns generator containing (child_key, child) for all children matching filter criteria
+        """
+        children = (
+            (child_key, self.store.get_item(child_key, depth=9))
```

Comment:
This block of code was moved from elsewhere in this file, and original author was @bradenmacdonald. So, @bradenmacdonald, could you please explain this to us, so I could add a comment?